test/cases/ptr_array_concat.zig

```diff
@@ -0,0 +1,21 @@
+const MyStruct = struct {
+    foobar: u8,
+};
+
+var my_struct1 = MyStruct{ .foobar = 1 };
+var my_struct2 = MyStruct{ .foobar = 2 };
+var my_struct3 = MyStruct{ .foobar = 3 };
+var my_struct4 = MyStruct{ .foobar = 4 };
+const foo: []const *MyStruct = &[_]*MyStruct{ &my_struct1, &my_struct2, &my_struct3, &my_struct4 };
+const bar: []const *MyStruct = foo[0..2] ++ foo[2..];
+
+pub fn main() void {
+    if (bar[0].*.foobar != 1 or
+        bar[1].*.foobar != 2 or
+        bar[2].*.foobar != 3 or
+        bar[3].*.foobar != 4)
+        unreachable;
```

Comment:
Andrew told me about this the other day but basically we should also prefer something like `std.testing.expect` over something that is basically `std.debug.assert` (which this is with `unreachable`):
```suggestion
    @import("std").testing.expect(bar[0].*.foobar == 1 and
        bar[1].*.foobar == 2 and
        bar[2].*.foobar == 3 and
        bar[3].*.foobar == 4);
```
This is so that it also tests properly in non-safe optimize mode.

Reply:
oh okay cool, I honestly didn't know you could use testing.expect outside of tests :sweat_smile:  updated again


btw I got the idea of just using `unreachable` from looking at some of the other tests in here like `adding_numbers_at_runtime_and_comptime.0.zig`

Reply:
Mhm yeah I also got the idea of using `std.debug.assert` from other tests but they really seem to be doing it wrong because well Andrew said this: https://github.com/ziglang/zig/pull/15879#pullrequestreview-1480744824

Reply:
wait, so it was actually correct to use `behavior` instead of `cases`, just not `behavior/bugs`?

Reply:
Yes, I think that's what he meant but I think the way you did it is acceptable as well.
I still used `cases` for a few tests which had multiple functions or needed global variables etc. because those kind of tests are harder to move to behavior test files.